targets/pna_nic/pna_nic.h

```diff
@@ -0,0 +1,173 @@
+/* Copyright 2024 Marvell Technology, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * Rupesh Chiluka (rchiluka@marvell.com)
+ *
+ */
+
+#ifndef PNA_NIC_PNA_NIC_H_
+#define PNA_NIC_PNA_NIC_H_
+
+#include <bm/bm_sim/queue.h>
+#include <bm/bm_sim/queueing.h>
+#include <bm/bm_sim/packet.h>
+#include <bm/bm_sim/switch.h>
+#include <bm/bm_sim/event_logger.h>
+#include <bm/bm_sim/simple_pre_lag.h>
+
+#include <memory>
+#include <chrono>
+#include <thread>
+#include <vector>
+#include <functional>
+
+// TODO(antonin)
+// experimental support for priority queueing
+// to enable it, uncomment this flag
+// you can also choose the field from which the priority value will be read, as
+// well as the number of priority queues per port
+// PRIORITY 0 IS THE LOWEST PRIORITY
+// #define NNIC_PRIORITY_QUEUEING_ON
+
+// #ifdef NNIC_PRIORITY_QUEUEING_ON
+// #define NNIC_PRIORITY_QUEUEING_NB_QUEUES 8
+// #define NNIC_PRIORITY_QUEUEING_SRC "intrinsic_metadata.priority"
+// #endif
```

Comment:
remove the code you don't need instead of commenting it out
even if you need it later, you can always add it back